src/NCMenuBar.cc

```diff
@@ -269,44 +279,117 @@ NCMenuBar::wHandleHotkey( wint_t key )
 }
 
 
-void
-NCMenuBar::select_next_menu()
+NCursesEvent
+NCMenuBar::handlePostMenu(const NCursesEvent & event, int selection)
 {
-    std::vector<Menu*>::iterator begin = _menus.begin();
+    NCursesEvent new_event = NCursesEvent::none;
 
-    if (_selected_menu)
+    if ( event == NCursesEvent::button && selection >= 0 )
     {
-	auto current = find(_menus.begin(), _menus.end(), _selected_menu);
+	YMenuItem * item = findMenuItem(selection);
 
-	if (current != _menus.end())
-	    begin = std::next(current, 1);
+	if ( item && item->isEnabled())
+	{
+	    new_event = NCursesEvent::menu;
+	    new_event.selection = item;
+	}
     }
+    else if ( event == NCursesEvent::key )
+    {
+	if ( event.keySymbol == "CursorLeft" )
+	{
+	    wHandleInput(KEY_LEFT);
+	    new_event = wHandleInput(KEY_DOWN);
+	}
+	else if ( event.keySymbol == "CursorRight" )
+	{
+	    wHandleInput(KEY_RIGHT);
+	    new_event = wHandleInput(KEY_DOWN);
+	}
+    }
+
+    return new_event;
+}
+
+
+void
+NCMenuBar::selectMenu( NCMenuBar::MenuIterator menu )
+{
+    if ( menu != _menus.end() )
+	_selected_menu = *menu;
+}
+
+
+NCMenuBar::MenuIterator
+NCMenuBar::currentMenu()
+{
+    return std::find(_menus.begin(), _menus.end(), _selected_menu);
+}
+
+
+NCMenuBar::MenuIterator
+NCMenuBar::nextMenu()
+{
+    MenuIterator current = currentMenu();
+
+    if ( current == _menus.end() )
+	return current;
+
+    MenuIterator begin = std::next( current, 1 );
 
-    auto next_menu = std::find_if(begin, _menus.end(), [](const Menu* menu) {
+    MenuIterator next = find_if(begin, _menus.end(), [](const Menu * menu) {
```

Comment:
I guess this tries to find the next menu item that is enabled, right?
Why not move that rather complicated expression out to a separate function `findNextEnabledMenuItem()` ?

Does it wrap around when it reaches the end? This seems to be the standard behaviour of other menu systems.